frontend/routes/notifications/notifications.component.tsx

```diff
@@ -0,0 +1,179 @@
+}
+
+const NotificationsDate = () =>
+	(<NotificationsPanelItem>
+			<NotificationsPanelHeader>
+				<ItemLabel>
+					{simpleDate(new Date())}
+				</ItemLabel>
+				<ItemLabel/>
+			</NotificationsPanelHeader>
+	</NotificationsPanelItem>);
+
+export class Notifications extends React.PureComponent<IProps, any> {
+	public state = {
+		open: false,
+		menuElement: null
+	};
+
+	public componentDidMount = () => {
+		// This will download notifications from the server and save to the store on init
+		this.props.sendGetNotifications();
+	}
+
+	public toggleDrawer = (e: React.SyntheticEvent) => {
+		this.setState(Object.assign({open: !this.state.open }));
+	}
+
+	public toggleMenu = (e: React.SyntheticEvent) => {
+		this.setState({ menuElement: this.state.menuElement ? null : e.currentTarget });
+		return false;
+	}
+
+	public deleteAllNotifications = (e: React.SyntheticEvent) => {
+		this.toggleMenu(e);
+		this.props.confirmSendDeleteAllNotifications();
+	}
+
+	public thisWeeksNotifications = (notifications) => {
+		const lastSunday = getSunday().getTime();
+		return notifications.filter((n) => n.timestamp > lastSunday);
+	}
+
+	public lastWeeksNotifications = (notifications) => {
+		const lastSunday = getSunday().getTime();
+		const prevSunday = getSunday(-1).getTime();
+		return notifications.filter((n) => n.timestamp > prevSunday && n.timestamp < lastSunday );
+	}
+
+	public moreThanTwoWeeksAgoNotifications = (notifications) => {
+		const prevSunday = getSunday(-1).getTime();
+		return notifications.filter((n) => n.timestamp < prevSunday );
+	}
+
+	public hasNotifications = () => {
+		return this.props.notifications.length > 0;
+	}
+
+	public renderNotificationsHeader = () => {
+		return (<ListSubheaderToolbar rightContent={
+					<>
+					<BarIconButton aria-label="Menu" style={{marginRight: -18}} onClick={this.toggleMenu}>
+						<Icon>more_vert</Icon>
+						<Menu
+							anchorEl={this.state.menuElement}
+							open={!!this.state.menuElement}
+							onClose={this.toggleMenu}
+							>
+								<MenuItem onClick={this.deleteAllNotifications} disabled={!this.hasNotifications()}>Delete all</MenuItem>
+							</Menu>
+					</BarIconButton>
+					<BarIconButton aria-label="Close panel" onClick={this.toggleDrawer}>
+						<Icon>close</Icon>
+					</BarIconButton>
+					</>
+					}>
+					<Typography variant={"title"} color={"inherit"} >
+						Notifications
+					</Typography>
+				</ListSubheaderToolbar>);
+	}
+
+	public render = () => {
+		const count =  this.props.notifications.filter((n) => !n.read).length;
+		const badgeColor = count > 0 ? "primary" : "secondary"; // Secondary color is used to make the badge disappear
+
+		const actions = {
+							sendUpdateNotificationRead: this.props.sendUpdateNotificationRead ,
+							sendDeleteNotification: this.props.sendDeleteNotification
+						};
+
+		const groupedByTeamspace = toArray(groupBy(sortBy(this.props.notifications, "teamSpace"), "teamSpace"));
+
+		return (
```

Comment:
I think you can split this render method to a few smaller. The code will be more readable.